rust-keybroker/keybroker-server/src/policy.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+use regorus::{self, Value};
+
+use crate::error::Result;
+
+// Evaluate an EAR claims-set against the appraisal policy and known-good RIM values
+pub(crate) fn rego_eval(
+    custom_policy: Option<String>,
+    rims: &str,
+    ear_claims: &str,
+) -> Result<Value> {
+    const POLICY: &str = include_str!("policy.rego");
+
+    // Create engine.
+    let mut engine = regorus::Engine::new();
+
+    engine.set_rego_v1(true);
+    engine.set_strict_builtin_errors(false);
+
+    // Add the appraisal policy
+    match custom_policy {
+        None => {
+            engine.add_policy(String::from("policy.rego"), String::from(POLICY))?;
+        }
+        Some(file) => {
+            engine.add_policy_from_file(file)?;
+        }
+    }
+
+    // Load the configured known good RIM values
+    engine.add_data(Value::from_json_file(rims)?)?;
+
+    // Set the EAR claims-set to be appraised
+    engine.set_input(Value::from_json_str(ear_claims)?);
+
+    let results = engine.eval_rule("data.arm_cca.allow".to_string())?;
```

Comment:
At the moment the calling convention is:

1. The policy package name is `"arm_cca"`
2. The policy output is conveyed through the `"allow"` variabile
3. The result of the policy evaluation is either `"true"` or `"false"`
4. The policy takes "some" reference values as input (*)

(*) When using the default policy, (4) requires a precise scheme:
```cddl
start = {
  "reference-values": [ + b64-rim ]
}

b64-rim = text .b64c rim

rim = bytes .size 32
```

I agree we could make the whole thing more flexible.

For example, we can allow customising the OPA rule, which is currently hardcoded to `"arm_cca.allow"` adding a new CLI switch (e.g., `--rule <...>`).
This would relax (1) and (2).

Regarding (4) we must:

* Document what is the expectation for the default policy, and (maybe)
* Provide a custom policy/ref-vals example.

I think (3) should stay as is, which is pretty reasonable.